src/lock/optiql.cpp

```diff
@@ -144,6 +159,10 @@ void
 RetainQID(  //
     const uint32_t qid)
 {
+  auto *qnode = &(_qnodes[qid]);
+  qnode->next.store(nullptr, kRelaxed);
+  qnode->ver.store(kNoLocks, kRelaxed);
```

Comment:
`QNode` の構成要素ですが，

```cpp
struct QNode {
  /// @brief The next queue node if exist.
  std::atomic<QNode *> next;

  /// @brief A flag for indicating this node's owner holds a lock.
  std::atomic_bool hold_lock;
};
```

としてください．これで初期状態（ゼロ埋め状態）でロックを握っていない状態になります．ここの返却時の初期化も

```suggestion
  qnode->next.store(nullptr, kRelaxed);
  qnode->hold_lock.store(false, kRelaxed);
```

ですね．